SYCL/InorderQueue/in_order_usm_host_dependency.cpp

```diff
@@ -0,0 +1,49 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// Test checks that dependency chain between commands is preserved for in-order
+// queue in the case when usm commands and host tasks are interleaved.
+
+#include <sycl.hpp>
+using namespace sycl;
+
+int main() {
+  queue Q{property::queue::in_order()};
+
+  int *X = malloc_host<int>(100, Q);
+  int *Y = malloc_host<int>(100, Q);
+  int *Z = malloc_host<int>(100, Q);
+
+  event MemSet = Q.memset(X, 0, sizeof(int) * 100);
+  MemSet.wait();
+  Q.submit([&](handler &CGH) {
+    auto HostTask = [=] {
+      for (int i = 0; i < 100; i++)
+        X[i] = 99;
+    };
+    CGH.host_task(HostTask);
+  });
+  Q.memset(Y, 99, sizeof(int) * 100);
+  Q.fill(X, 98, 100);
+  Q.memcpy(Y, X, sizeof(int) * 100);
+  Q.copy(Y, Z, 100);
+
+  Q.submit([&](handler &CGH) {
+    auto HostTask = [=] { Z[99] *= 2; };
+    CGH.host_task(HostTask);
+  });
+  Q.wait();
+
+  int Error = (Z[99] != 196) ? 1 : 0;
```

Comment:
I guess it should check if host task waits till usm operation completion but could we also check the reverse case? I see that you fill X with values using host task and than overwrite it many times with usm operations but never check and result in final check is also not dependent on it.

Reply:
Good call, fixed.